### ISSUE TITLE
Support libxml++ 3.0.0+

### ISSUE THREAD

**Issue by Quintus** [Secretchronicles/TSC#591]
Hi,

currently, we do not support libxml++ >= 3.0.0 for TSC. The problem with this is that the CMake module for finding libxml++ fails to work with that version of libxml (it simply doesn't detect it), i.e. there is a bug in CMake. Until this is resolved, we cannot support libxml++ >= 3.0.0. I will file a bug with CMake later on; appearently it is confused by the renaming of the library due to ABI change. I'll link the issue here once I get around to do that.

Valete,
Quintsu

**Comment by Quintus** [comment]
Okay, for detecting libxml++ we actually have [our own CMake module](https://github.com/Secretchronicles/TSC/blob/29bc0c6bad5ff49f055ccc083636f1ee9f72e289/tsc/cmake/modules/FindLibXmlPP.cmake) which is at fault. It is pretty easy to adapt to also find libxml++-3.0 by just replacing all occurences of "2.6" with "3.0".

However, appearently libxml++'s API has changed quite a bit, and trying to support both APIs clutters the code with lots of `#ifdef` statements, since it is used at many places (I just tried to compile TSC against libxml++-3.0). So, we have to explicitely figure out when to switch to the new API.

Suggestion: Once `libxml++-3.0` lands in an Ubuntu LTS, we upgrade this dependency. @xet7, if you get a chance, could you look up in which (LTS) Ubuntu version `libxml++` >= 3.0.0 is going to be included?

Below is a patch against commit c06f1380ecb67a4f6c91ca405a9429eda8c321d1 suitable for `git am` to update the CMake module for finding libxml++ in case I forget how I did this.

Valete,
Quintus

~~~~ patch
From 86827321307bdb1f7470bb65b525a4ea211b2fa9 Mon Sep 17 00:00:00 2001
From: Quintus <quintus@quintilianus.eu>
Date: Sat, 14 Jan 2017 20:49:09 +0100
Subject: [PATCH] Change FindLibXmlPP cmake module to find libxml++3.0.

---
 tsc/cmake/modules/FindLibXmlPP.cmake | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tsc/cmake/modules/FindLibXmlPP.cmake b/tsc/cmake/modules/FindLibXmlPP.cmake
index 60e40be..9a42af5 100644
--- a/tsc/cmake/modules/FindLibXmlPP.cmake
+++ b/tsc/cmake/modules/FindLibXmlPP.cmake
@@ -14,14 +14,14 @@
 
 pkg_check_modules(PKG_LibXmlPP_Glib QUIET glib-2.0)
 pkg_check_modules(PKG_LibXmlPP_GlibMM QUIET glibmm-2.4)
-pkg_check_modules(PKG_LibXmlPP QUIET libxml++-2.6)
+pkg_search_module(PKG_LibXmlPP QUIET libxml++-3.0)
 
 # Note we use a subdirectory find (libxml++/document.h) since that is what
 # we include. Many other libs may have a document.h file, so one definitely
 # does not want that subdirectory into the compiler include search path
 # (prevents conflicts).
 find_path(LibXmlPP_INCLUDE_DIR libxml++/document.h
-  HINTS ${PKG_LibXmlPP_INCLUDE_DIRS} /usr/include/libxml++-2.6)
+  HINTS ${PKG_LibXmlPP_INCLUDE_DIRS} /usr/include/libxml++-3.0)
 
 # libxml++'s headers include glibmm headers, which is irritating.
 # When including libxml++'s headers, these need to be in the compiler
@@ -44,11 +44,11 @@ find_path(LibXmlPP_Glib_Config_INCLUDE_DIR glibconfig.h
 
 # libxml++'s headers also include a config header.
 find_path(LibXmlPP_Config_INCLUDE_DIR libxml++config.h
-  HINTS ${PKG_LibXmlPP_INCLUDE_DIRS} /usr/lib/libxml++-2.6/include)
+  HINTS ${PKG_LibXmlPP_INCLUDE_DIRS} /usr/lib/libxml++-3.0/include)
 
 # Now for the actual libraries.
 find_library(LibXmlPP_LIBRARY
-  NAMES xml++ xml++-2.6
+  NAMES xml++ xml++-3.0
   HINTS ${PKG_LibXmlPP_LIBRARY_DIRS})
 find_library(LibXmlPP_GlibMM_LIBRARY
   NAMES glibmm-2.4 glibmm
-- 
2.10.2
~~~~

**Comment by Quintus** [comment]
Implemented by 8a5fd591786b9a7621951c777bd5b6f81988bcd0 via PR #612.
